### ISSUE TITLE
TrueNAS Scale 23.10 ISCSI: Cannot mount new pvc after NIC change on k8s nodes

### ISSUE THREAD

**Issue by oscgu** [democratic-csi/democratic-csi#379]
We upgraded from onboard NICs to dedicated ones and now newly created pvs cannot be mounted on our k8s nodes.
The truenas server is reachable from the node and the underlying volume of the pv exists.
Weirdly enough only Volumes that existed prior to the NIC change still work.

Logs of the csi-driver when trying to mount a newly created pvc:
```
grpc implementation: @grpc/grpc-js
{"host":"worker-01","level":"info","message":"initializing csi driver: freenas-api-iscsi","service":"democratic-csi","timestamp":"2024-03-26T18:31:43.958Z"}
{"host":"worker-01","level":"info","message":"starting csi server - node version: v16.18.0, package version: 1.8.4, config file: /config/..2024_03_26_18_31_42.2153066161/driver-config-file.yaml, csi-name: org.democratic-csi.iscsi-truenas-06, csi-driver: freenas-api-iscsi, csi-mode: node, csi-version: 1.5.0, address: , socket: unix:///csi-data/csi.sock.internal","service":"democratic-csi","timestamp":"2024-03-26T18:31:44.358Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: GetPluginInfo call: {\"metadata\":{\"x-forwarded-host\":[\"localhost\"],\"user-agent\":[\"grpc-go/1.54.0\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:05.513Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: GetPluginInfo response: {\"name\":\"org.democratic-csi.iscsi-truenas-06\",\"vendor_version\":\"1.8.4\"}","service":"democratic-csi","timestamp":"2024-03-26T18:32:05.513Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: NodeGetInfo call: {\"metadata\":{\"user-agent\":[\"grpc-go/1.54.0\"],\"x-forwarded-host\":[\"localhost\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:06.038Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: NodeGetInfo response: {\"node_id\":\"worker-01\",\"max_volumes_per_node\":0}","service":"democratic-csi","timestamp":"2024-03-26T18:32:06.038Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: NodeGetInfo call: {\"metadata\":{\"user-agent\":[\"grpc-go/1.54.0\"],\"x-forwarded-host\":[\"localhost\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.411Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: NodeGetInfo response: {\"node_id\":\"worker-01\",\"max_volumes_per_node\":0}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.411Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: NodeGetCapabilities call: {\"metadata\":{\"x-forwarded-host\":[\"localhost\"],\"user-agent\":[\"grpc-go/1.54.0\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.746Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: NodeGetCapabilities response: {\"capabilities\":[{\"rpc\":{\"type\":\"STAGE_UNSTAGE_VOLUME\"}},{\"rpc\":{\"type\":\"GET_VOLUME_STATS\"}},{\"rpc\":{\"type\":\"EXPAND_VOLUME\"}},{\"rpc\":{\"type\":\"SINGLE_NODE_MULTI_WRITER\"}}]}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.746Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: NodeGetCapabilities call: {\"metadata\":{\"user-agent\":[\"grpc-go/1.54.0\"],\"x-forwarded-host\":[\"localhost\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.763Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: NodeGetCapabilities response: {\"capabilities\":[{\"rpc\":{\"type\":\"STAGE_UNSTAGE_VOLUME\"}},{\"rpc\":{\"type\":\"GET_VOLUME_STATS\"}},{\"rpc\":{\"type\":\"EXPAND_VOLUME\"}},{\"rpc\":{\"type\":\"SINGLE_NODE_MULTI_WRITER\"}}]}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.764Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: NodeGetCapabilities call: {\"metadata\":{\"user-agent\":[\"grpc-go/1.54.0\"],\"x-forwarded-host\":[\"localhost\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.765Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: NodeGetCapabilities response: {\"capabilities\":[{\"rpc\":{\"type\":\"STAGE_UNSTAGE_VOLUME\"}},{\"rpc\":{\"type\":\"GET_VOLUME_STATS\"}},{\"rpc\":{\"type\":\"EXPAND_VOLUME\"}},{\"rpc\":{\"type\":\"SINGLE_NODE_MULTI_WRITER\"}}]}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.766Z"}
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: NodeStageVolume call: {\"metadata\":{\"user-agent\":[\"grpc-go/1.54.0\"],\"x-forwarded-host\":[\"localhost\"]},\"request\":{\"publish_context\":{},\"secrets\":\"redacted\",\"volume_context\":{\"portal\":\"192.168.1.30:3260\",\"portals\":\"\",\"provisioner_driver\":\"freenas-api-iscsi\",\"storage.kubernetes.io/csiProvisionerIdentity\":\"1711408936989-8682-org.democratic-csi.iscsi-truenas-06\",\"interface\":\"\",\"iqn\":\"iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster\",\"lun\":\"0\",\"node_attach_driver\":\"iscsi\"},\"volume_id\":\"pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7\",\"staging_target_path\":\"/var/lib/kubelet/plugins/kubernetes.io/csi/org.democratic-csi.iscsi-truenas-06/ba2fb71390bedd42c8515123f55a674534cab425bab25ff61e92923a6201497b/globalmount\",\"volume_capability\":{\"access_mode\":{\"mode\":\"SINGLE_NODE_MULTI_WRITER\"},\"mount\":{\"mount_flags\":[],\"fs_type\":\"ext4\",\"volume_mount_group\":\"\"},\"access_type\":\"mount\"}},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:07.773Z"}
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -o new
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -o update --name node.startup --value manual
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
{"host":"worker-01","level":"info","message":"new request - driver: FreeNASApiDriver method: Probe call: {\"metadata\":{\"user-agent\":[\"grpc-node-js/1.9.9\"]},\"request\":{},\"cancelled\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:09.570Z"}
{"host":"worker-01","level":"info","message":"new response - driver: FreeNASApiDriver method: Probe response: {\"ready\":{\"value\":true}}","service":"democratic-csi","timestamp":"2024-03-26T18:32:09.570Z"}
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
retry - waiting 2000ms before trying again
executing iscsi command: iscsiadm -m node -T iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster -p 192.168.1.30:3260 -l
{"host":"worker-01","level":"error","message":"handler error - driver: FreeNASApiDriver method: NodeStageVolume error: {\"code\":19,\"stdout\":\"Logging in to [iface: default, target: iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster, portal: 192.168.1.30,3260]\\n\",\"stderr\":\"iscsiadm: Could not login to [iface: default, target: iqn.2005-10.org.freenas.ctl:csi-pvc-f63b809b-5058-48ed-a4dd-179b6eaa33d7-cluster, portal: 192.168.1.30,3260].\\niscsiadm: initiator reported error (19 - encountered non-retryable iSCSI login failure)\\niscsiadm: Could not log into all portals\\n\",\"timeout\":false}","service":"democratic-csi","timestamp":"2024-03-26T18:32:36.725Z"}
```
It's probably trying to use the old NIC for some calls?
Neither rebooting the node nor restarting all the democratic-csi pods helped so far.

K8s version: 1.28.2
democratic-csi version: helm 0.14.5
TrueNAS Scale version: 23.10

**Comment by travisghansen** [comment]
Welcome! As a starter I would check the `scst` logs on the SCALE server to see if that gives any indication why the login/connection is failing. Give that a try first and let's see what it is reporting and then we can decide next steps.

**Comment by oscgu** [comment]
Thank you! I checked the logs using `journalctl -xeu scst` and this is all I get (when trying to mount again):
![asdf](https://github.com/democratic-csi/democratic-csi/assets/94227101/551234cf-67fd-44df-bd71-1faed19e0951)

**Comment by travisghansen** [comment]
Not super helpful is it. Can you try attaching to the iscsi target from another node which has nothing to do with the cluster and make sure that works?

**Comment by oscgu** [comment]
I tried to mount the new pvc on a different host, outside of the cluster, but it's not even showing up as a target when I do `iscsiadm -m discovery -t sendtargets -p 192.168.1.30`.
All the previously created pvcs/targets show up though.
Additionally I manually created a new zvol, configured the iscsi stuff and it also does not show up on the client side.
So I suppose it's an issue with our TrueNAS server after all...
Any ideas what I could do, apart from restarting the server or what else to look into?

**Comment by travisghansen** [comment]
Well, if you review the assets in the TN ui do they appear? If so, go right to the console and cat out the scst config file and ensure they translated from the TN DB to the actual running config file.

**Comment by oscgu** [comment]
I checked `/etc/scst.conf` and the assets are there and appear correct, when compared with existing ones.
Maybe restarting the scst service would help? Unsure though how disruptive that would be, for existing connections

**Comment by travisghansen** [comment]
Hmm, is access being blocked by policy on the portal group(s)? If not I would recommend reaching out ix and getting some eyes on it from that side.

**Comment by oscgu** [comment]
Yeah not an policy issue either... I opened a ticket on their jira now, thank you for the help so far!

**Comment by oscgu** [comment]
It was a regression introduced by fixing another issue (wrt ALUA in a HA environment).
The bug happens when you remove a target and then add a new one.

They have fixed it now, but looks like it will ship with the TrueNAS 24.* releases ([Jira Ticket](https://ixsystems.atlassian.net/jira/software/c/projects/NAS/issues/NAS-128099))

**Comment by travisghansen** [comment]
Thanks!
